### PR TITLE
[Docs] Distinguish warm vs cold vs frozen tiers

### PR DESCRIPTION
**Pull request by Leaf-Lin**
Today, the description at the top of this data tier page can be confusing for users on how to choose between warm vs cold or cold vs frozen. 

I added some descriptions around cold and frozen tiers to emphasise the ability of use of searchable snapshot and their storage benefit based on https://www.elastic.co/elasticsearch/elasticsearch-searchable-snapshots and https://www.elastic.co/blog/introducing-elasticsearch-frozen-tier-searchbox-on-s3. Feel free to reword it.

### CODE REVIEW

File: docs/reference/datatiers.asciidoc
Comment (dakrone) [inline docs/reference/datatiers.asciidoc:13]: I think this implies that it happens automatically, so I would recommend something like:
```suggestion
* <<cold-tier, Cold tier>> nodes hold time series data that is accessed infrequently and not normally updated. <<fully-mounted,Fully mounted indices>> of <<ilm-searchable-snapshot,{search-snaps}>> may be kept on the cold tier, eliminating the need for replicas with approximately 50% storage reduction compared to the warm tier.
```

File: docs/reference/datatiers.asciidoc
Comment (dakrone) [inline docs/reference/datatiers.asciidoc:14]:
```suggestion
* <<frozen-tier, Frozen tier>> nodes hold time series data that is accessed rarely and never updated. <<partially-mounted,Partially mounted indices>> of <<ilm-searchable-snapshot,{search-snaps}>> are kept on frozen tier to extend the storage capacity even further by up to 20 times compare to the warm tier.
```

**Review by dakrone — COMMENTED** [review]
I left some wording recommendations but it would be good for @jrodewig to take a look also

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [reply docs/reference/datatiers.asciidoc]: One thing I never grasp with English grammar is `tense`.  

In the previous [commit](https://github.com/elastic/elasticsearch/commit/5c97e845db636d3df4dd116c5519b4a2c0512e73#r809549540), you propose to change `compares` to `compared`.  In this sentence, you propose to change `compares` to `compare`.  

What have I missed? :P

Comment (dakrone) [reply docs/reference/datatiers.asciidoc]: I had a typo, they should both be `compared`, I fixed my typo now :)

File: docs/reference/datatiers.asciidoc
Comment (jrodewig) [inline docs/reference/datatiers.asciidoc:13]: Two suggestions:

- I'd point out that the storage savings is in disk space.
- I'd point out that you can use regular indices on the cold tier if wanted.

```suggestion
* <<cold-tier,Cold tier>> nodes hold time series data that is accessed
infrequently and not normally updated. To save space, you can keep
<<fully-mounted,fully mounted indices>> of
<<ilm-searchable-snapshot,{search-snaps}>> on the cold tier. These fully mounted
indices eliminate the need for replicas, reducing required disk space by
approximately 50% compared to the regular indices.
+
Alternatively, you can the cold tier to store regular indices with replicas instead
of using {search-snaps}. This lets you store older data on less expensive hardware
but doesn't reduce required disk space compared to the warm tier.
```

**Review by jrodewig — COMMENTED** [review]
Thanks @Leaf-Lin. I left a few comments and suggestions I think we should address before merging.

Let me know if you have any questions or if I can help in any way.

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [inline docs/reference/datatiers.asciidoc:88]:
```suggestion
the warm tier.  It can also be used for storing read-only
```

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [reply docs/reference/datatiers.asciidoc]: Great suggestion. Could you check if my latest commit satisfies the above?

I made quite a lot of modifications in the cold tier paragraph. Mainly to point out that one can use cold tier just like warm tier, or use searchable snapshot. 

```
Cold tier can be used for storing rarely updated time series data similar to 
the warm tier.  It can also be used for storing read-only
<<fully-mounted,fully mounted indices>> of
<<ilm-searchable-snapshot,{search-snaps}>>.  When one chooses to use searchable
snapshot on the cold tier, it eliminates the need for replicas with
approximately 50% storage reduction compared to the warm tier.
```

This is indeed a frequent question users are confused about the cold tier.

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [inline docs/reference/datatiers.asciidoc:87]:
```suggestion
Cold tier can be used for storing infrequently updated time series data similar to 
```
Change to infrequently to be consistent with other places.

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [reply docs/reference/datatiers.asciidoc]: I have modified the paragraph and used the word `exclusively`, hope that can justify explicitly mention?

```
* <<frozen-tier, Frozen tier>> nodes hold time series data that is accessed 
rarely and never updated. It is necessary to have a snapshot repository setup
in order to use frozen tier. The frozen tier stores
<<partially-mounted,Partially mounted indices>> of
<<ilm-searchable-snapshot,{search-snaps}>> exclusively.  This extends the
storage capacity even further by up to 20 times compare to the warm tier. 
```

Additionally, I have also seen quite a lot of confusion around frozen node setup where users may not realize it depends on a snapshot repository. So I added the sentence to emphasize the need.

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [inline docs/reference/datatiers.asciidoc:17]:
```suggestion
* <<cold-tier,Cold tier>> nodes hold time series data that is accessed
infrequently and not normally updated. To save space, you can keep
<<fully-mounted,fully mounted indices>> of
<<ilm-searchable-snapshot,{search-snaps}>> on the cold tier. These fully mounted
indices eliminate the need for replicas, reducing required disk space by
approximately 50% compared to the regular indices.
+
Alternatively, you can use the cold tier to store regular indices with replicas instead
of using {search-snaps}. This lets you store older data on less expensive hardware
but doesn't reduce required disk space compared to the warm tier.
```

File: docs/reference/datatiers.asciidoc
Comment (Leaf-Lin) [reply docs/reference/datatiers.asciidoc]: 🤦 , I somehow missed this suggestion while I was working my way through the above.  I have added this commit in https://github.com/elastic/elasticsearch/pull/84155#discussion_r813521447 and in the main body.
 
To avoid duplication, I'd prefer to simplify the cold tier description in the summary at the top, and put the detailed description in the content, but I felt all of these points are important though, um...

Additionally, I am reading https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html and felt that some of the descriptions around cold or frozen data **nodes** can be updated accordingly or linked to content on this page.  Can you assist with that please?

File: docs/reference/datatiers.asciidoc
Comment (jrodewig) [inline docs/reference/datatiers.asciidoc:110]: Nit: rewording
```suggestion
The frozen tier requires a snapshot repository.
```

**Review by jrodewig — COMMENTED** [review]
Thanks @Leaf-Lin. I left some suggestions — mostly to deduplicate the content. If my suggestions look okay to you, I think this is good to go.

> Additionally, I am reading https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-node.html and felt that some of the descriptions around cold or frozen data nodes can be updated accordingly or linked to content on this page. Can you assist with that please?

Sure thing. I'll handle that in a separate PR.

Edit: I raised a draft PR here: https://github.com/elastic/elasticsearch/pull/84346. That gives you an idea of where I'm heading, but I think we can do better at content reuse. That'll be easier to plumb once these changes are merged in.

**Comment by Leaf-Lin** [comment]
@jrodewig .  LGTM, thanks for your suggestions! I think this still needs approval before merging?

**Review by jrodewig — APPROVED** [review]
LGTM. I'll merge this in.

**Comment by jrodewig** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.1|[](https://github.com/elastic/elasticsearch/pull/84435)|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/84436)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/84437)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**
